bids/layout/layout.py

```diff
@@ -381,6 +381,25 @@ def collect_layouts(layout):
         layouts = [l for l in collect_layouts(self) if l._in_scope(scope)]
         return list(set(layouts))
 
+    def _sanitize_query_dtypes(self, entities):
+        """ Automatically convert entity query values to correct dtypes. """
+        entities = entities.copy()
+        names = list(entities.keys())
+        ents = {e.name: e for e in
+                self.session.query(Entity)
+                    .filter(Entity.name.in_(names)).all()}
+            # Fail silently because the DB may still know how to reconcile
+            # type differences.
+        for name, val in entities.items():
+            try:
+                if isinstance(val, (list, tuple)):
```

Comment:
The problem with checking iterables is that strings are iterables, and we definitely don't want to iterate strings. This is a more general issue that we should probably address by creating a package-wide `is_iterable` helper that checks for any iterable but a string. There are other places throughout the codebase where we'd need to make the same update.

That said, this has never been a problem in practice, so I don't consider it a high priority. `get()` is not really a place where users should be passing around iterables other than lists or tuples anyway, since you wouldn't know you could do that unless you read the documentation.

Reply:
ugh, why doesn't string have an as_type method.
What about using `type(val)(ents[name])`?

```
try:
    filters[nv] = type(val)(filters[nv])
except TypeError:
    try:
        filters[nv] = [type(val)(rr) for rr in filters[nv]]
    except:
        pass
except:
    pass
```

Reply:
sorry, that code is using the variable names from my pr, do you get what I mean though.

Reply:
I don't like this so much because the results are going to be much less predictable. E.g., if you pass a `dict` in as the value (which as far as I'm concerned should just fail outright), the sanitization could succeed, and you'd get back, e.g., a list of the keys of the dict. I'd rather just enforce lists or tuples, and if you pass anything else, you either haven't read the docs, or you've made a mistake.

Reply:
Fair enough, it'd be good to raise an error if its some iterable that's not a list or a tuple, but that can be dealt with later. Thanks for addressing the bug so quickly.